packages/flutter_driver/lib/src/driver.dart

```diff
@@ -94,7 +97,9 @@ class FlutterDriver {
   /// Creates a driver that uses a connection provided by the given
   /// [_serviceClient], [_peer] and [_appIsolate].
   @visibleForTesting
-  FlutterDriver.connectedTo(this._serviceClient, this._peer, this._appIsolate);
+  FlutterDriver.connectedTo(this._serviceClient, this._peer, this._appIsolate,
+      { bool printCommunication: false, bool logCommunicationToFile: true })
+      : _printCommunication = printCommunication, _logCommunicationToFile = logCommunicationToFile;
```

Comment:
As discussed in person: debugging (especially when dealing with random flakes) will be a lot easier if the log is always produced.

To avoid the multi-device testing problem, I added an ID to each driver instance and each instance now logs to flutter_driver_commands_${driverId}.log.